File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.py
import math
def solution(begin, end):
    answer = []
    
    def find(n):
        if n == 1:
            return 0
        data = []
        for i in range(2, int(n**0.5)+1):
            if n%i == 0:
                data.append(i)
                if n//i <=10000000:
                    return n//i
        if len(data)>=1:
            return data[-1]
        return 1
        
    for i in range(begin,end+1):
        answer.append(find(i))
    
    return answer

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.java
import java.util.*;
class Solution {
    public int[] solution(long begin, long end) {
        int be = (int)begin;
        int en = (int)end;
        int[] answer = new int[en-be+1];
        Arrays.fill(answer,1);
        for(int i=be;i<en+1;i++){
            answer[i-be] = calc(i);
        }
        if(be==1) answer[0] = 0;
        return answer;
    }
    public static int calc(int start){
        int max = 1;
        for(int i=2; i*i<=start;i++){
            if(start%i==0){
                max = i;
                if(start/i<= 10000000)
                return start/i;
            }
        }
        return max;
    }
}